src/main/java/mate/academy/springbootintro/service/shoppingcart/impl/ShoppingCartServiceImpl.java

```diff
@@ -35,53 +48,76 @@ public ShoppingCartDto findShoppingCardByUserId(Long id) {
     @Override
     @Transactional
     public ShoppingCartDto addCartItem(
-            CreateCartItemRequestDto cartItemRequestDto,
```

Comment:
See the example of implementation method addCartItem

```
Book book = bookRepository.findById(itemDto.bookId())
                .orElseThrow(() -> new EntityNotFoundException(
                        String.format("Book with id: %d is not found", itemDto.bookId())
                ));
        ShoppingCart cart = shopCartRepository.findByUserId(userId);
        cart.getCartItems().stream()
                .filter(item -> item.getBook().getId().equals(itemDto.bookId()))
                .findFirst()
                .ifPresentOrElse(item -> item.setQuantity(item.getQuantity() + itemDto.quantity()),
                        () -> addCartItemToCart(itemDto, book, cart));
        shopCartRepository.save(cart);
        return cartMapper.toDto(cart);
```

Where 
```
private void addCartItemToCart(CartItemRequestDto itemDto, Book book, ShoppingCart cart) {
        CartItem cartItem = itemMapper.toCartItem(itemDto);
        cartItem.setBook(book);
        cart.addItemToCart(cartItem);
    }
```

And in ShoppingCart model

```
public void addItemToCart(CartItem cartItem) {
        cartItem.setShoppingCart(this);
        cartItems.add(cartItem);
    }
```